src/Abstractions/NexusMods.Abstractions.Loadouts/Mods/Mod.cs

```diff
@@ -17,85 +23,172 @@ namespace NexusMods.Abstractions.Loadouts.Mods;
+    
     /// <summary>
-    /// Version of the mod
+    /// The Download Metadata the mod was installed from.
     /// </summary>
-    public string Version { get; set; } = string.Empty;
-
+    public static readonly ReferenceAttribute Source = new(Namespace, nameof(Source));
+    
     /// <summary>
-    /// Name of the mod in question.
+    /// The enabled status of the mod
     /// </summary>
-    public required string Name { get; init; } = string.Empty;
-
+    public static readonly BooleanAttribute Enabled = new(Namespace, nameof(Enabled));
+    
     /// <summary>
-    /// True if the mod is enabled, false otherwise.
+    /// The install status of the mod.
     /// </summary>
-    public bool Enabled { get; init; } = false;
-
-    /// <inheritdoc />
-    public override EntityCategory Category => EntityCategory.Mods;
-
+    public static readonly EnumAttribute<ModStatus> Status = new(Namespace, nameof(Status));
+    
     /// <summary>
-    /// Defines the individual sorting rules applied to a game.
+    /// The category of the mod.
     /// </summary>
-    public ImmutableList<ISortRule<Mod, ModId>> SortRules { get; init; } = ImmutableList<ISortRule<Mod, ModId>>.Empty;
-
+    public static readonly EnumAttribute<ModCategory> Category = new(Namespace, nameof(Category)) { IsIndexed = true };
+    
+    
     /// <summary>
-    /// The install status of the mod.
+    /// Sort this mod after another mod, mostly used as a placeholder until we figure out better
+    /// sorting mechanisms.
     /// </summary>
-    public ModStatus Status { get; init; } = ModStatus.Installed;
+    public static readonly ReferenceAttribute SortAfter = new(Namespace, nameof(SortAfter));
+    
 
     /// <summary>
-    /// Date and time when the mod was installed.
+    /// Gets all the revisions of a loadout over time
     /// </summary>
-    public DateTime Installed { get; set; } = DateTime.UtcNow;
+    public static IObservable<Model> Revisions(this IConnection conn, ModId id)
+    {
+        // All db revisions that contain the loadout id, select the loadout
+        return conn.Revisions
+            .Where(db => db.Datoms(db.BasisTxId).Any(datom => datom.E == id.Value))
+            .StartWith(conn.Db)
+            .Select(db => db.Get<Model>(id.Value));
+    }
+
+
+    public partial class Model(ITransaction tx) : Entity(tx)
+    {
+        
+        /// <summary>
+        /// Remaps the entity id into a mod id, this is mostly just a cast.
+        /// </summary>
+        public ModId ModId => ModId.From(Id);
+        
+        public string Name
+        {
+            get => Mod.Name.Get(this);
+            set => Mod.Name.Add(this, value);
+        }
+        
+        /// <summary>
+        /// The revision number for this mod, increments by one for each change.
+        /// </summary>
+        public ulong Revision
+        {
+            get => Mod.Revision.Get(this, 0);
+            set => Mod.Revision.Add(this, value);
+        }
```

Comment:
Same issue as Loadout revision.

Reply:
Fixed this by removing one overload of "revision Id", LoadoutRevisionId is now "LoadoutWithTxId". We still need to improve this a bit, but what we have now is a good starting point.